src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,141 @@
+import cn from 'classnames';
+import {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo
+  onUpdate: (todo: Todo, prop: string, value: any) => void
+  onDelete: (todoId: number) => void
+  changingTodoIds: number | number[]
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  onUpdate,
+  onDelete,
+  changingTodoIds,
+}) => {
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(false);
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isEditing]);
+
+  const titleEditHandler = () => {
+    if (!newTitle.length) {
+      onDelete(todo.id);
+    } else if (newTitle !== todo.title) {
+      onUpdate(todo, 'title', newTitle);
+    }
+
+    setIsEditing(false);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', {
+        completed: todo.completed,
+      })}
+      key={todo.id}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          onClick={() => {
+            onUpdate(
+              todo,
+              'completed',
+              !todo.completed,
+            );
+          }}
+          type="checkbox"
+          className="todo__status"
+          defaultChecked
+        />
+      </label>
+
+      {isEditing
+        ? (
+          <form
+            action="submit"
+            onSubmit={(event) => {
+              event.preventDefault();
+              titleEditHandler();
+            }}
+          >
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              key={todo.id}
+              value={newTitle}
+              onDoubleClick={() => {
+                setIsEditing(true);
+              }}
+              onChange={(event) => {
+                setNewTitle(event.target.value);
+              }}
+              onKeyDown={(event) => {
+                if (event.key === 'Escape') {
+                  setIsEditing(false);
+                  setNewTitle(todo.title);
+                }
+              }}
+              ref={newTodoField}
+              onBlur={titleEditHandler}
+              className="todo__title-field"
+              placeholder="What needs to be done?"
+              disabled={!isEditing}
+            />
+          </form>
+        )
+        : (
+          <>
+            <span
+              key={todo.id}
```

Comment:
I couldn't understand why there is a key here. Todoinfo components is being assigned a key in the todolist while mapping.